apps/prairielearn/elements/pl-symbolic-input/pl-symbolic-input.js

```diff
@@ -0,0 +1,353 @@
+      } else if (ev.key === 'Escape') {
+        ev.preventDefault();
+      }
+    },
+    { capture: true },
+  );
+
+  // Workaround for glitchy size-based menu button show/hide behavior
+  if (!mf.getAttribute('read-only')) {
+    const resizeObserver = new ResizeObserver((_) => {
+      if (mf.offsetWidth < 160) {
+        mf.setAttribute('hide-menu', 'true');
+      } else {
+        mf.removeAttribute('hide-menu');
+      }
+    });
+    resizeObserver.observe(mf);
+  }
+};
+
+/**
+ * Initialize the <math-field> element used to represent parse errors.
+ * This element is read-only and only needs to be updated with the set of allowed function names for display
+ *
+ * @param {string} uuid UUID of the element to look up the tag (used instead of name since there might be multiple submissions per element)
+ */
+window.PLSymbolicInputParseError = async function (uuid) {
+  const mf = document.getElementById(`symbolic-input-parse-error-${uuid}`);
+  if (!mf) {
+    return;
+  }
+  setUpSymbolicInputMacros(mf);
+};
+
+/**
+ * Initialize the <math-field> element used to represent raw parsed submissions.
+ * This element is read-only and only needs to be updated with the set of allowed function names for display
+ *
+ * @param {string} uuid UUID of the element to look up the tag (used instead of name since there might be multiple submissions per element)
+ */
+window.PLSymbolicInputPopover = async function (uuid) {
+  // Wait for popover to appear
+  $(document).on('shown.bs.popover', `#pl-symbolic-input-${uuid}-button`, function () {
+    const mf = document.getElementById(`symbolic-input-popover-${uuid}`);
+    if (!mf) {
+      return;
+    }
+    setUpSymbolicInputMacros(mf);
+  });
+};
+
+/**
+ * Initialize list of allowed function names for a <math-field> element. The list is determined by
+ * the allow-trig and custom-functions attributes and each allowed function is added as a "macro" to
+ * automatically replace individual letters with well-formatted, atomic function name blocks.
+ *
+ * @param {HTMLElement} mf The <math-field> element to initialize
+ */
+function setUpSymbolicInputMacros(mf) {
+  const additionalFunctions = mf.getAttribute('custom-functions')
+    ? mf.getAttribute('custom-functions').split(',')
+    : [];
+
+  // Note that we include trigonometric functions in the macros, even if they are disabled for grading
+  // Otherwise, an accidental function call like "sin" would be converted into "s*i*n", which leads to
+  // a much harder to understand error message (about "s" being an unknown variable) generated by SymPy.
+  const customFunctions = new Set(
+    additionalFunctions.concat([
+      'Abs',
+      'abs',
+      'acos',
+      'acosh',
+      'arccos',
+      'arcsin',
+      'arctan',
+      'arctan2',
+      'asin',
+      'asinh',
+      'atan',
+      'atan2',
+      'atanh',
+      'cos',
+      'cot',
+      'csc',
+      'exp',
+      'factorial',
+      'ln',
+      'log',
+      'Max',
```

Comment:
It looks like you have an uppercase version of a couple of these. Why is that needed and what is the criteria for that?